File: 백준/Silver/10815. 숫자 카드/숫자 카드.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {
	static int n,m;
	static int arr[];
	public static void main(String args[]) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		
		
		n = Integer.parseInt(br.readLine());
		 arr = new int [n];
		 String input_1 [] = br.readLine().split(" ");
		 
		for(int i = 0; i < n ; i++) {
			arr[i] = Integer.parseInt(input_1[i]);
		}
		Arrays.sort(arr);
		
		m = Integer.parseInt(br.readLine());
		String input_2 [] = br.readLine().split(" ");
		 
		
		for(int i = 0 ; i < m; i++) {
			int num = Integer.parseInt(input_2[i]);
			if(binarySearch(num)) {
				System.out.print("1 ");
				
			}else {
				System.out.print("0 " );
			}
		}
		
	}

	public static boolean binarySearch(int num) {
		
		int left = 0;
		int right = n-1;
		while(left <= right) {
			int midindex = (left + right)/2;
			int mid = arr[midindex];
			if(num < mid) {
				right = midindex -1;
			}else if(num > mid) {
				left = midindex +1;
			}
			else return true;
		}
		return false;
	}
}

File: 백준/Silver/10815. 숫자 카드/숫자 카드.swift
import Foundation

func Q_10815() {
    let _ = Int(readLine()!)!
    let aryN = readLine()!.split { $0 == " " }.map { Int($0)! }.sorted()
    let _ = Int(readLine()!)!
    let aryM = readLine()!.split { $0 == " " }.map { Int($0)! }

    for m in aryM {
        print(BinarySearch(aryN, m))
    }

    func BinarySearch(_ ary: [Int], _ key: Int) -> Int {
        var start = 0
        var end = ary.count
        var mid = 0

        while end - start > 0 {
            mid = (start + end) / 2
            if ary[mid] < key {
                start = mid + 1
            } else if ary[mid] > key {
                end = mid
            } else {
                return 1
            }
        }

        return 0
    }
}
Q_10815()